File: coffee-system/src/main/java/com/jesperancinha/coffee/system/objects/EmployeeLayer.java
package com.jesperancinha.coffee.system.objects;

import org.jesperancinha.coffee.system.input.CoffeeMachines.CoffeMachine.Coffees.Coffee;
import org.jesperancinha.coffee.system.input.CoffeeMachines.CoffeMachine.PaymentTypes.Payment;
import org.jesperancinha.coffee.system.input.Employees;
import lombok.Builder;
import lombok.Getter;

/**
 * Created by joaofilipesabinoesperancinha on 05-05-16.
 */
@Builder
@Getter
public class EmployeeLayer {
    private Employees.Employee employee;

    private Coffee coffee;

    private Payment payment;
}

File: coffee-system/src/main/kotlin/org/jesperancinha/coffee/system/objects/EmployeeLayer.kt
package org.jesperancinha.coffee.system.objects

import org.jesperancinha.coffee.system.input.CoffeeMachines.CoffeMachine.Coffees.Coffee
import org.jesperancinha.coffee.system.input.CoffeeMachines.CoffeMachine.PaymentTypes.Payment
import org.jesperancinha.coffee.system.input.Employees.Employee

/**
 * Created by joaofilipesabinoesperancinha on 05-05-16.
 */
data class EmployeeLayer(
    val employee: Employee,
    private val coffee: Coffee? = null,
    private val payment: Payment? = null,
)